src/data.jl

```diff
@@ -224,6 +224,34 @@ function Base.getindex(data::Data, inds, var::Regex)
   getindex(data, inds, snames)
 end
 
+Base.vcat(data::Data...) = reduce(_vcat, data)
+
+function _vcat(data1, data2)
+  tab1 = values(data1)
+  tab2 = values(data2)
+  dom1 = domain(data1)
+  dom2 = domain(data2)
+
+  cols1 = Tables.columns(tab1)
+  cols2 = Tables.columns(tab2)
+  names = Tables.columnnames(cols1)
+  
+  if Set(names) ≠ Set(Tables.columnnames(cols2))
+    throw(ArgumentError("All data must have the same variables"))
+  end
```

Comment:
This is not necessarily the case. As we discussed before, the DataFrames.jl package supports the concatenation even when some columns are not present via the option cols=:union (if I remember correctly).

Reply:
I remember this. But my intention was to start with a simpler implementation. Then we can add more features in future PRs.

Reply:
Makes total sense. Good strategy, just need to communicate better the plan.